src/Tribe/Views/V2/Widgets/Admin_Template.php

```diff
@@ -27,10 +49,152 @@ public function __construct() {
+		}
+	}
+
+	/**
+	 * Section templating to handle control/input sections and fieldsets.
+	 *
+	 * @since TBD
+	 *
+	 * @param int                  $field_id    The ID of the field.
+	 * @param array <string,mixed> $field       The field info.
+	 * @param array                $passthrough Passthrough data (from parent - like fieldset, to children).
+	 */
+	public function section( $field_id, $field, $passthrough = [] ) {
+		$data = [
+			'id'         => $this->widget_obj->get_field_id( $field_id ),
+			'name'       => $this->widget_obj->get_field_name( $field_id ),
+			'label'      => Arr::get( $field, 'label', '' ),
+			'options'    => Arr::get( $field, 'options', [] ),
+			'value'      => Arr::get( $this->context, $field_id, [] ),
+			'classes'    => Arr::get( $field, 'classes', '' ),
+			'children'   => Arr::get( $field, 'children', '' ),
+			'dependency' => $this->format_dependency( $field ),
+		];
+
+		$this->template( "widgets/components/{$field['type']}", $data, $field, $passthrough );
+	}
+
+	/**
+	 * Input templating.
+	 *
+	 * @since TBD
+	 *
+	 * @param int                  $field_id    The ID of the field.
+	 * @param array <string,mixed> $field       The field info.
+	 * @param array                $passthrough Passthrough data (from parent - like fieldset, to children).
+	 */
+	public function input( $field_id, $field, $passthrough = [] ) {
+		$data = [
+			'id'          => $this->widget_obj->get_field_id( $field_id ),
+			'name'        => $this->widget_obj->get_field_name( $field_id ),
+			'label'       => Arr::get( $field, 'label', '' ),
+			'options'     => Arr::get( $field, 'options', [] ),
+			'value'       => Arr::get( $this->context, $field_id, [] ),
+			'classes'     => Arr::get( $field, 'classes', '' ),
+			'placeholder' => Arr::get( $field, 'placeholder', '' ),
+			'dependency'  => $this->format_dependency( $field ),
+		];
+
+		if ( 'radio' === $field['type'] ) {
+			$data[ 'button_value' ] = Arr::get( $field, 'button_value', '' );
+			$data['name']           = Arr::get( $passthrough, 'name', '' );
+			$data['value']          = Arr::get( $passthrough, 'value', null );
+		}
+
+		if ( in_array( $field['type'], $this->allowed_field_types ) ) {
+			$this->template( "widgets/components/{$field['type']}", $data );
+		} else {
+			/**
+			 * Allows injection of custom "inputs" by other plugins.
+			 *
+			 * @since TBD
+			 *
+			 * @param array<string,mixed> $data       The field data passed to templates.
+			 * @param array<string,mixed> $field      The field info.
+			 * @param WP_Widget           $widget_obj The widget object.
+			 */
+			do_action( "tribe_events_view_v2_widget_admin_form_{$field['type']}_input", $data, $field, $this->widget_obj, $this->context );
+		}
+	}
+
+	/**
+	 * Massages the data before asking tribe_format_field_dependency() to create the dependency attributes.
+	 *
+	 * @since TBD
+	 *
+	 * @param array <string,mixed> $field The field info.
+	 * @return void
```

Comment:
:no_entry_sign: Function return type is void, but function contains return statement

<sub>hash: a0beb12f85b8e39ba2b3cc9c4db78e57</sub>